com.unity.netcode.gameobjects/Runtime/Messaging/Messages/ConnectionRequestMessage.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace Unity.Netcode.Messages
+{
+    internal struct ConnectionRequestMessage : INetworkMessage
+    {
+        public ulong ConfigHash;
+
+        [StructLayout(LayoutKind.Explicit, Size = 512)]
+        public struct ConnectionDataStorage : IFixedArrayStorage
+        {
+
+        }
+
+        public FixedUnmanagedArray<byte, ConnectionDataStorage> ConnectionData;
```

Comment:
can we use `NativeArray<byte>` for this?

Reply:
The difference between NativeArray and FixedUnmanagedArray here is that FixedUnmanagedArray 1) is entirely stack memory, doesn't even need UnsafeUtility.Malloc, and 2) has a flexible length. The thing that's fixed is the runtime memory usage. But you can make a FixedUnmanagedArray<byte, 512> and put only 2 bytes in it and it will serialize only two bytes... if you did that with NativeArray, it would serialize 512 bytes.

Reply:
we could use different allocators for NativeArray (temp).
do we need flexible length here?
can't we serialize n bytes + count?

Reply:
NativeArray doesn't allow a count separate from its capacity. NativeList does, but doesn't allow you to specify a fixed capacity that won't grow. But it's ok, I'm already removing FixedUnmanagedArray.

Reply:
It will be removed in this PR ?

Reply:
Removed... just using byte[] directly for now, since it ultimately has to be a byte[] to be passed to InvokeConnectionApproval anyway.